airflow/api/common/experimental/mark_tasks.py

```diff
@@ -181,7 +181,39 @@ def set_state(task, execution_date, upstream=False, downstream=False,
         if len(sub_dag_ids) > 0:
             tis_altered += qry_sub_dag.all()
 
+    session.expunge_all()
     session.close()
 
     return tis_altered
 
+def set_dag_run_state(dag, execution_date, state=State.SUCCESS, commit=False):
```

Comment:
Actually, put this in models.py. It's okay to have the stub here but the logic of setting the dagrun to the state should be in models